eogrow/utils/validators.py

```diff
@@ -91,6 +91,18 @@ def ensure_both(cls: type, value: Optional[Any], values: RawSchemaDict) -> Optio
     return field_validator(second_param, ensure_both, **kwargs)
 
 
+def ensure_storage_key_presence(key: str, **kwargs: Any) -> classmethod:
+    """A field validator that makes sure that the specified storage key is present in the storage structure."""
+
+    def validate_storage_key(cls: type, key: Optional[str], values: RawSchemaDict) -> Optional[str]:
+        if key is not None:
+            storage_keys = list(values["storage"].structure.keys()) + ["input_data"]
```

Comment:
what about logs and cache?

you can also drop the `keys` btw. I'd even do something like `assert key in storage.structure or key in predefined_keys` (with apropriate variables)

Reply:
wasn't sure if those are expected as folder keys. do you think there is a way to get the preset folders of the storage manager so that I don't cause a circular import?? or is the hardcoding here acceptable?

Reply:
i'd say hardcoding is ok, cause i really dont know where else to put this. We might want to reevaluate the structure

Reply:
updated